From 200ac31b901a191bc620bc0dbdd7e2d44193de8a Mon Sep 17 00:00:00 2001
From: Nickey Yang <nickey.yang@rock-chips.com>
Date: Wed, 7 Nov 2018 16:43:12 +0800
Subject: [PATCH] regulator: act8865: make pm_power_off can be defined.

change to use act8865_power_off instead of psci_sys_poweroff
if the property "system-power-controller" is set.

Change-Id: Ie3a46255419488063b57b2829da2ae8f15f014ae
Signed-off-by: Nickey Yang <nickey.yang@rock-chips.com>
---
 drivers/regulator/act8865-regulator.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/regulator/act8865-regulator.c b/drivers/regulator/act8865-regulator.c
index f8d4cd3d1397..6e9592c2e1e5 100644
--- a/drivers/regulator/act8865-regulator.c
+++ b/drivers/regulator/act8865-regulator.c
@@ -504,13 +504,13 @@ static int act8865_pmic_probe(struct i2c_client *client,
 	}
 
 	if (of_device_is_system_power_controller(dev->of_node)) {
-		if (!pm_power_off && (off_reg > 0)) {
+		if (off_reg > 0) {
 			act8865_i2c_client = client;
 			act8865->off_reg = off_reg;
 			act8865->off_mask = off_mask;
 			pm_power_off = act8865_power_off;
 		} else {
-			dev_err(dev, "Failed to set poweroff capability, already defined\n");
+			dev_err(dev, "Failed to set poweroff capability\n");
 		}
 	}
 
-- 
2.35.3

